src/main/java/org/folio/edge/sip2/domain/messages/requests/EndPatronSession.java

```diff
@@ -0,0 +1,152 @@
+package org.folio.edge.sip2.domain.messages.requests;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+/**
+ * Represents an End Patron Session message.
+ *
+ * <blockquote
+ *     cite="http://multimedia.3m.com/mws/media/355361O/sip2-protocol.pdf">
+ * This message will be sent when a patron has completed all of their
+ * transactions. The ACS may, upon receipt of this command, close any open
+ * files or deallocate data structures pertaining to that patron. The ACS
+ * should respond with an End Session Response message.
+ * </blockquote>
+ *
+ * @author mreno-EBSCO
+ *
+ */
+public final class EndPatronSession {
+  /** The date of this transaction. */
+  private final ZonedDateTime transactionDate;
+  /** The ID of the institution making the request. */
+  private final String institutionId;
```

Comment:
Some fields could, probably at most it'd be `transactionId` and `institutionId`, but even those are not in every message. I couldn't come up with a reasonable hierarchy for inheritance, mostly because it would require multiple inheritance to do it right. We might be able to use composition to do this. We may want to investigate it at some point.